### PR TITLE
5 get articles

### PR DESCRIPTION
**Pull request by Garyismatic**
Created and tested /api/articles endpoint that responds with an array of objects with all available articles including their:

Title, 
Article id,
Author,
Votes,
Topic,
Date Created,
Article Image,
Number of Comments.

They are sorted by date created as default with newly created first.

### CODE REVIEW

File: __tests__/api.test.js [hunk 1]
```diff
@@ -43,6 +43,45 @@ describe('/api/topics', () => {
...
+                    })
+                })
+            })
+        })
+        it('returns all of the articles from the db', () => {
+            return request(app)
+            .get('/api/articles')
+            .expect(200)
+            .then(({body: {articles}}) => {
+                expect(articles.length).toBe(13)
```
Comment (Harrison365) [inline __tests__/api.test.js:72]: Please merge the forEach and length check tests, as one request may be working and the other not.

File: __tests__/api.test.js [hunk 2]
```diff
@@ -43,6 +43,45 @@ describe('/api/topics', () => {
...
+            })
+        })
+        it('returns the articles sorted by date newly created first', () => {
+            return request(app)
+            .get('/api/articles')
+            .expect(200)
+            .then(({body: {articles}}) => {
+                expect(articles).toBeSortedBy('created_at', {descending: true})
+            })
+        })
```
Comment (Harrison365) [inline __tests__/api.test.js:82]: nice work 🙂

File: models/articles.model.js [hunk 3]
```diff
@@ -1,5 +1,13 @@
 const db = require('../db/connection')
 
+exports.fetchArticles = () => {
+
+    return db.query('SELECT COUNT (comments.body) AS comment_count, articles.author, title, articles.article_id, topic, articles.created_at, articles.votes, article_img_url FROM articles FULL OUTER JOIN comments ON comments.article_id = articles.article_id  GROUP BY articles.article_id ORDER BY created_at DESC')
+    .then(({rows}) => {
+        return rows
+    })
+}
+
```
Comment (Harrison365) [inline models/articles.model.js:10]: Nice :)

File: controllers/articles.controller.js [hunk 4]
```diff
@@ -1,4 +1,14 @@
-const { fetchArticleById } = require("../models/articles.model")
+const { fetchArticleById, fetchArticles } = require("../models/articles.model")
+
+exports.getArticles = (request, response, next) => {
+    return fetchArticles()
+    .then((articles) => {
+        articles.forEach((article) => {
+            article.comment_count = parseInt(article.comment_count) //<< change comment_count to number from string
+        })
```
Comment (Harrison365) [inline controllers/articles.controller.js:8]: You can achieve this using CAST() in your SQL. You may want to refactor instead of using a loop here.

Comment (Garyismatic) [reply controllers/articles.controller.js:8]: I'm not 100% clear on where the CAST would fit into my SQL string

**Pull request merged**